crates/model/src/solver_competition.rs

```diff
@@ -74,6 +76,9 @@ pub struct CompetitionAuction {
 pub struct SolverSettlement {
     pub solver: String,
     pub objective: Objective,
+    #[serde(with = "ratio_as_decimal")]
+    pub score: BigRational, // CIP17 score
```

Comment:
I think this should be `U256` as it represents an Ether value in wei.

At least this is my interpretation of the CIP.

Reply:
Went with `f64` since the `score` have the same semantics as rewards, because it represents the amount of reward the solver is willing to give up for the submitted solution. Since rewards are `f64` then `f64` makes most sense probably.

Reply:
I'm unrsolving this to double check that Nick is ok with f64. I'm still not sure.

Reply:
I'm unrsolving this to double check that Nick is ok with f64. I'm still not sure.

Reply:
@nlordell pinging in case Nick missed the message.

Reply:
> same semantics as rewards

To me, they have different semantics. One is a CoW value (whose computation in the CIP is defined for floating point numbers, and not integers). The score is an Ether amount (and I think we should use either `wei: U256` or `units: BigDecimal`.

Reply:
@vkgnosis makes a valid point that `BigDecimal` is unbounded numerical value. I think we should use `U256` for this as a wei amount in Ether.